templates/etc/haproxy/defaults.cfg.j2

```diff
@@ -27,3 +27,9 @@
   errorfile {{ errorfile.code }} {{ errorfile.file }}
 {% endfor %}
 {% endif %}
+
+{% if haproxy_defaults_compression != false %}
```

Comment:
I rather disable compression by default, can you change it to?

`{% for compression in haproxy_defaults_compression | default([]) %}`

and remove the section from `defaults/main.yml`. The rest of the changes look good. Thanks in advance!